src/extension/csw/core/src/main/java/org/geoserver/csw/store/internal/CatalogStoreFeatureCollection.java

```diff
@@ -90,4 +105,55 @@ protected void closeIterator(Iterator<Feature> close) {}
         MemoryFeatureCollection memory = new MemoryFeatureCollection(getSchema(), features);
         return memory.sort(order);
     }
+
+    private Filter catalogFilter() {
+
+        // ignore catalog info's without id
+        Filter result =
+                ff.and(filter, ff.not(ff.isNull(mapping.getIdentifierElement().getContent())));
+
+        // build filter compatible with layergroups and resources
+        result =
```

Comment:
Can you split this code a bit, so that it does not end up being nested so much? Creating a separate inner class would probably do the trick